sympy/testing/randtest.py

```diff
@@ -4,6 +4,7 @@
 
 from sympy.core.containers import Tuple
 from sympy.core.numbers import comp, I
+from sympy.core.singleton import S
```

Comment:
The testing module should ideally not import anything from the rest of sympy except potentially from utilities. Since I moved the testing module here I have been meaning to move this submodule out. It depends on core and so should not be part of testing.

Reply:
So should randtest.py go back to utilities?

Reply:
I don't think it should be in utilities either since I don't think that utilities should import anything from core. Probably only `_randint` and `_randrange` belong in utilities although I'm not convinced that either is needed at all.

Maybe the rest should go into the `sympy.core.random` or something. The only place any of this is used in non-testing code is in `sympy.core.expr`.

See also #16967

Reply:
> although I'm not convinced that either is needed at all.

The routines were provided as a convenience to allow a routine to control the numbers being provided by the same function that would otherwise be selecting them on its own based on the value of seed.

```python
>>> r = _randrange(range(10))
>>> [r(10) for i in range(10)]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]

>>> r=_randrange([2,3,2,1,4,6,5])
>>> while 1:
...   r(3,5)
...
3
4
Traceback (most recent call last):
...
ValueError: _randrange sequence was too short
```